pyvcloud/vcd/vdc.py

```diff
@@ -415,6 +426,12 @@ def update_disk(self, name, size, new_name=None, description=None,
         else:
             disk = self.get_disk(name)
 
+        if storage_profile_name is not None:
+            storage_profile = self.get_storage_profile(storage_profile_name)
+            sp = etree.XML(etree.tostring(storage_profile, pretty_print=True))
+            sp_href = sp.attrib['href']
+            diskParms.append(E.StorageProfile(href=sp_href,
+                                              name=storage_profile_name))
         if disk is None:
```

Comment:
Move line 424-427 and 435-436 to the top of the function. Flow of the function should be
1. Get the existing disk object. If any error occurs here, we should bail as soon as possible.
2. Construct the new disk param
3. Make the rest call.

Reply:
I will change this in another CL, since this does not break anything.